### PR TITLE
Incorporating the Voucher feed for Finance

### PR DESCRIPTION
**Pull request by carolyncole**
This PR includes the original code from @mzelesky along with some refactoring to get it into classes.

It still needs some work to get rubocop to pass.

It also needs to connect to the FTP server to get the alma file and do something with the output file.

### CODE REVIEW

**Comment by mzelesky** [comment]
I think the address info for the sample xml from Alma needs to be scrubbed.

File: spec/models/alma_xml_invoice_spec.rb [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+
+    context "invalid invoice" do
+      let(:xml_file) { File.new(Rails.root.join('spec', 'fixtures', 'invalid_invoice.xml')) }
+
+      it "is not valid" do
+        expect(alma_invoice.valid?).to be_falsey
+        expect(alma_invoice.errors).to contain_exactly("Invalid vendor_id; vendor_id can not be blank.",
+                                                       "Invalid primary fund; can not be blank.",
+                                                       "Invalid primary department; can not be blank.",
+                                                       "Invalid reporting code; must numeric and can not be blank.")
```
Comment (kevinreiss) [inline spec/models/alma_xml_invoice_spec.rb:111]: "must be numeric"

File: spec/models/alma_xml_invoice_list_spec.rb [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+  let(:sftp_entry1) { instance_double("Net::SFTP::Protocol::V01::Name", name: "abc.xml") }
+  let(:sftp_entry2) { instance_double("Net::SFTP::Protocol::V01::Name", name: "abc.xml2") }
+  let(:sftp_entry3) { instance_double("Net::SFTP::Protocol::V01::Name", name: "123.xml") }
+  let(:sftp_session) { instance_double("Net::SFTP::Session", dir: sftp_dir) }
+  let(:sftp_dir) { instance_double("Net::SFTP::Operations::Dir") }
+
+  let(:xml_file) { File.new(Rails.root.join('spec', 'fixtures', 'invoice_export_202118300518.xml')) }
+  let(:invoice_errors) do
+    "Invalid vendor_id; vendor_id can not be blank.,"\
```
[the remark this answers is not in the document]
Comment (carolyncole) [reply spec/models/alma_xml_invoice_list_spec.rb:15]: Kevin, it is a comma separated list of errors. I should remove the period.  Good catch

**Comment by carolyncole** [comment]
@kevinreiss Thank you for the feedback!  I've updated the PR.

**Review by kevinreiss — APPROVED** [review]

**Pull request merged**